codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-auth-scheme-interceptor.java

```diff
@@ -0,0 +1,54 @@
+package software.amazon.awssdk.services.query.rules.internal;
+
+import java.util.List;
+import java.util.function.Supplier;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.auth.signer.Aws4Signer;
+import software.amazon.awssdk.auth.signer.SignerLoader;
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.rules.AwsEndpointAttribute;
+import software.amazon.awssdk.awscore.rules.AwsEndpointProviderUtils;
+import software.amazon.awssdk.awscore.rules.EndpointAuthScheme;
+import software.amazon.awssdk.awscore.util.SignerOverrideUtils;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.core.interceptor.SdkInternalExecutionAttribute;
+import software.amazon.awssdk.core.rules.model.Endpoint;
+import software.amazon.awssdk.core.signer.Signer;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+public final class QueryEndpointAuthSchemeInterceptor implements ExecutionInterceptor {
```

Comment:
Have you considered adding this logic as a part of the `AwsExecutionContextBuilder` between initial interceptors and signers instead of using an interceptor? I can't see any service-specific values at the moment.

Reply:
`signerProvider` is service specific :)

Reply:
It took me a while, I was just looking at the generated file and not the spec :-). Ok.